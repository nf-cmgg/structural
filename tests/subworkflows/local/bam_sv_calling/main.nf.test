nextflow_workflow {

    name "Test Workflow BAM_SV_CALLING"
    script "subworkflows/local/bam_sv_calling/main.nf"
    workflow "BAM_SV_CALLING"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_sv_calling"

    options "-stub"

    config "./nextflow.config"

    test("homo_sapiens - manta") {

        when {
            params {
                callers = "manta"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - delly") {

        when {
            params {
                callers = "delly"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    test("homo_sapiens - smoove") {

        when {
            params {
                callers = "smoove"
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

    // test("homo_sapiens - gridss") {

    //     when {
    //         params {
    //             callers = "gridss"
    //         }
    //         workflow {
    //             """
    //             input[0] = Channel.of([
    //                 [id:"test", sample:"test", sex:"male"],
    //                 file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
    //                 file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
    //             ])
    //             input[1] = Channel.value([
    //                 [id:"fasta"],
    //                 file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
    //             ])
    //             input[2] = Channel.value([
    //                 [id:"fai"],
    //                 file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
    //             ])
    //             input[3] = [[],[]]
    //             input[4] = []
    //             """
    //         }
    //     }

    //     then {
    //         assertAll(
    //             { assert workflow.success },
    //             { assert snapshot(
    //                 workflow.out.reports,
    //                 workflow.out.vcfs,
    //                 workflow.trace.succeeded().size(),
    //                 workflow.trace.failed().size()
    //             ).match() }
    //         )
    //     }

    // }

    test("homo_sapiens - all sv callers") {

        when {
            params {
                callers = "manta,delly,smoove" //,gridss
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
                ])
                input[3] = [[],[]]
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.reports,
                    workflow.out.vcfs,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
