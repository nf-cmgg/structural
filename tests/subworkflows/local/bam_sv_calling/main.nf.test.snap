{
    "homo_sapiens - smoove": {
        "content": [
            [
                
            ],
            [
                {
                    "id": "test",
                    "sample": "test",
                    "sex": "male",
                    "variant_type": "sv"
                },
                "test.svync.smoove.vcf.gz",
                "test.svync.smoove.vcf.gz.tbi"
            ],
            5,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-04-29T13:24:56.132319567"
    },
    "homo_sapiens - manta": {
        "content": [
            [
                
            ],
            [
                {
                    "id": "test",
                    "sample": "test",
                    "sex": "male",
                    "variant_type": "sv"
                },
                "test.svync.manta.vcf.gz",
                "test.svync.manta.vcf.gz.tbi"
            ],
            6,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:04:24.17658191"
    },
    "homo_sapiens - delly": {
        "content": [
            [
                
            ],
            [
                {
                    "id": "test",
                    "sample": "test",
                    "sex": "male",
                    "variant_type": "sv"
                },
                "test.svync.delly.vcf.gz",
                "test.svync.delly.vcf.gz.tbi"
            ],
            3,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:04:41.148774402"
    },
    "homo_sapiens - all sv callers": {
        "content": [
            [
                
            ],
            [
                {
                    "id": "test",
                    "sample": "test",
                    "sex": "male",
                    "variant_type": "sv"
                },
                "test.sv.vcf.gz",
                "test.sv.vcf.gz.tbi"
            ],
            18,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.02.0"
        },
        "timestamp": "2024-03-11T14:05:53.52450825"
    }
}